tests/components/hydrawise/test_binary_sensor.py

```diff
@@ -0,0 +1,45 @@
+"""Test Hydrawise binary_sensor."""
+
+from datetime import timedelta
+from unittest.mock import Mock
+
+from homeassistant.components.hydrawise.const import SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.util.dt import utcnow
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_states(
+    hass: HomeAssistant, mock_added_config_entry: MockConfigEntry
+) -> None:
+    """Test binary_sensor states."""
+    # Make the coordinator refresh data.
+    async_fire_time_changed(hass, utcnow() + SCAN_INTERVAL + timedelta(seconds=30))
```

Comment:
```suggestion
async def test_states(
    hass: HomeAssistant,
    mock_added_config_entry: MockConfigEntry,
    freezer: FrozenDateTimeFactory,
) -> None:
    """Test binary_sensor states."""
    # Make the coordinator refresh data.
    freezer.tick(SCAN_INTERVAL + timedelta(seconds=30))
    async_fire_time_changed(hass)
```
Freezer seems to be more stable with changing time in tests